hpOneView/resources/servers/enclosures.py

```diff
@@ -331,3 +331,64 @@ def get_utilization(self, id_or_uri, fields=None, filter=None, refresh=False, vi
             dict
         """
         return self._client.get_utilization(id_or_uri, fields=fields, filter=filter, refresh=refresh, view=view)
+
+    def generate_csr(self, csr_data, id_or_uri, bay_number=None):
+        """
+        Creates a Certificate Signing Request (CSR) for an enclosure.
+
+        Args:
+            csr_data: Dictionary with csr details.
+            id_or_uri: Can be either the resource ID or the resource URI.
+            bay_number: OA from which a the CSR should be generated.
```

Comment:
from which **a the**
probably drop either of those

Reply:
Not sure why you have not added default value for Type: EnclosureV7. Is it intentional?

Reply:
updated